ledger/src/blockstore.rs

```diff
@@ -170,6 +172,48 @@ impl<T> AsRef<T> for WorkingEntry<T> {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct LastFECSetCheckResults {
+    is_full: bool,
+    is_retransmitter_signed: bool,
+}
+
+impl LastFECSetCheckResults {
+    pub fn report_metrics(&self, slot: Slot, hash: Hash) {
+        if !self.is_full {
+            datapoint_warn!(
+                "incomplete_final_fec_set",
+                ("slot", slot, i64),
+                ("hash", hash.to_string(), String)
+            );
+        }
+        if !self.is_retransmitter_signed {
+            datapoint_warn!(
+                "invalid_retransmitter_signature_final_fec_set",
+                ("slot", slot, i64),
+                ("hash", hash.to_string(), String)
+            );
+        }
+    }
+
+    pub fn get_result(
+        &self,
+        feature_set: &FeatureSet,
+    ) -> std::result::Result<(), BlockstoreProcessorError> {
+        if feature_set.is_active(&solana_sdk::feature_set::vote_only_full_fec_sets::id())
+            && !self.is_full
+        {
+            return Err(BlockstoreProcessorError::IncompleteFinalFecSet);
+        } else if feature_set
+            .is_active(&solana_sdk::feature_set::verify_retransmitter_signature::id())
```

Comment:
I am reusing the the verification flag here. I believe that should be fine, however I can create a separate flag if desired.

Reply:
Yeah, we probably want  a separate `vote_only_...` feature just in case.